configure.ac

```diff
@@ -44,8 +44,11 @@ AS_IF([test "x$found_introspection" = xyes],
       [AC_SUBST(WITH_GI, 1)],
       [AC_SUBST(WITH_GI, 0)])
 
-uname -p|grep s390
-on_s390=$?
+AC_CANONICAL_HOST
+case $host_cpu in
+  s390|s390x) on_s390=0 ;;
+  *) on_s390=1 ;;
```

Comment:
The use of 0 (true) and 1 (false) is a bit confusing here.
It would be more readable if you use ```on_s390=yes|no``` imho

Reply:
Especially considering a line like this:
```AS_IF([test "$on_s390" = "1" -a "x$with_s390" = "xyes"],```

Reply:
I did want do keep the same values as before (return code of the `uname | grep`) but this is definitely better. Fixed.